src/sentry/api/endpoints/auth_index.py

```diff
@@ -38,28 +38,26 @@
 
 
 @control_silo_endpoint
-class AuthIndexEndpoint(Endpoint):
-    publish_status = {
-        "DELETE": ApiPublishStatus.UNKNOWN,
-        "GET": ApiPublishStatus.UNKNOWN,
-        "PUT": ApiPublishStatus.UNKNOWN,
-        "POST": ApiPublishStatus.UNKNOWN,
-    }
+class BaseAuthIndexEndpoint(Endpoint):
     """
-    Manage session authentication
-
-    Intended to be used by the internal Sentry application to handle
-    authentication methods from JS endpoints by relying on internal sessions
-    and simple HTTP authentication.
+    Base endpoint to manage session authentication. Shared between
+    AuthIndexEndpoint and StaffAuthIndexEndpoint (in getsentry)
     """
 
     owner = ApiOwner.ENTERPRISE
     authentication_classes = (QuietBasicAuthentication, SessionAuthentication)
 
     permission_classes = ()
 
+    def get(self, request: Request) -> Response:
+        if not request.user.is_authenticated:
+            return Response(status=status.HTTP_400_BAD_REQUEST)
```

Comment:
Instead of bad request here, could we return `unauthorized` (aka `401`)? https://www.django-rest-framework.org/api-guide/status-codes/#client-error-4xx
400 is misleading, as it's not a bad request in itself, but rather that an aspect of the request is in correct; which in this case, is the authorization part.

I understand this might mean changing how the error code is handled on the FE side, but we should capture this somewhere and update it in another PR. Our goal should be to send codes that help the user determine what action they should take to resolve the issue on their request (as for all endpoints and responses).

Reply:
Even if the users in this case is us (since it's private methods) haha

Reply:
We could take this a step further and make this behavior more standard, and prevent having to repeat this logic everywhere by setting up a middleware or decorator that does this. Is that something we already have that we could hook into?

Reply:
I think I can return `401` without needing to change the FE. We can some logic that I believe does the check if you are using permission classes, but since we dont have any on the endpoint we need it manually here.